code/game/objects/items/mail.dm

```diff
@@ -0,0 +1,267 @@
+	var/special_name = FALSE
+
+	if(prob(25))
+		special_name = TRUE
+		junk = pick(list(/obj/item/paper/pamphlet/gateway, /obj/item/paper/pamphlet/violent_video_games, /obj/item/paper/fluff/junkmail_redpill, /obj/effect/decal/cleanable/ash))
+
+	var/list/junk_names = list(
+		/obj/item/paper/pamphlet/gateway = "[initial(name)] for [pick(GLOB.adjectives)] adventurers",
+		/obj/item/paper/pamphlet/violent_video_games = "[initial(name)] for the truth about the arcade centcom doesn't want to hear",
+		/obj/item/paper/fluff/junkmail_redpill = "[initial(name)] for those feeling [pick(GLOB.adjectives)] working at nanotrasen",
+		/obj/effect/decal/cleanable/ash = "[initial(name)] with INCREDIBLY IMPORTANT ARTIFACT- DELIVER TO SCIENCE DIVISION",
+	)
+
+	color = pick(department_colors) //eh, who gives a shit.
+	name = special_name ? junk_names[junk] : "important [initial(name)]"
+
+	junk = new junk(src)
+	return TRUE
+
+/// Subtype that's always junkmail
+/obj/item/mail/junkmail/Initialize()
+	..()
+	junk_mail()
+
+/// Crate for mail from CentCom.
+/obj/structure/closet/crate/mail
+	name = "mail crate"
+	desc = "A certified post crate from CentCom."
+	icon_state = "mail"
+
+/// Crate for mail that automatically generates a lot of mail. Usually only normal mail, but on lowpop it may end up just being junk.
+/obj/structure/closet/crate/mail/full
+	name = "mail crate"
+	desc = "A certified post crate from CentCom."
+	icon_state = "mail"
+
+/obj/structure/closet/crate/mail/update_icon_state()
+	. = ..()
+	if(opened)
+		icon_state = "[initial(icon_state)]open"
+		for(var/obj/item/mail/Mail in src)
+			icon_state = initial(icon_state)
+			break
+	else
+		icon_state = "[initial(icon_state)]sealed"
+
+/obj/structure/closet/crate/mail/full/Initialize()
+	. = ..()
+	var/list/mail_recipients = list()
+	for(var/mob/living/carbon/human/alive in GLOB.player_list)
+		if(alive.stat != DEAD)
+			mail_recipients += alive
+	for(var/iterator in 1 to 22)
+		var/obj/item/mail/New_mail
+		if(prob(70))
+			New_mail = new /obj/item/mail(src)
+		else
+			New_mail = new /obj/item/mail/envelope(src)
+		var/mob/living/carbon/human/mail_to
+		if(mail_recipients.len)
+			mail_to = pick(mail_recipients)
+		if(prob(50)) //so after 21 passes if everyone's at least gotten something we'll junkmail it up
+			mail_recipients -= mail_to
+		if(mail_to)
+			New_mail.initialize_for_recipient(mail_to)
+		else
+			New_mail.junk_mail()
+
+/// Mailbag.
+/obj/item/storage/bag/mail
+	name = "mail bag"
+	desc = "A bag for letters, envelopes, and other postage."
+	icon = 'icons/obj/library.dmi'
+	icon_state = "bookbag"
+	worn_icon_state = "bookbag"
+	resistance_flags = FLAMMABLE
+
+/obj/item/storage/bag/mail/ComponentInitialize()
+	. = ..()
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+	STR.max_w_class = WEIGHT_CLASS_NORMAL
+	STR.max_combined_w_class = 42
+	STR.max_items = 21
+	STR.display_numerical_stacking = FALSE
+	STR.set_holdable(list(
+		/obj/item/mail
```

Comment:
desc implies it can carry regular cargo packages i.e. wrapped boxes, why not let it carry that and boxes?